sql: cleanup SURVIVE ... FAILURE syntax

otan opened the pull request:
sql: remove SURVIVE DEFAULT; rename Survive to SurvivalGoal

We still need a concept of "default" for tree.SurviveDefault so that we
do not display it syntactically, but the value will be transformed
before it is persisted.

Release note: None

sql: rename SURVIVE AVAILABILITY ZONE FAILURE to SURVIVE ZONE FAILURE

Release note (sql change): SURVIVE AVAILABILITY ZONE FAILURE is now
SURVIVE ZONE FAILURE.

otan committed:
sql: rename SURVIVE AVAILABILITY ZONE FAILURE to SURVIVE ZONE FAILURE

Release note (sql change): SURVIVE AVAILABILITY ZONE FAILURE is now
SURVIVE ZONE FAILURE.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/56837)
<!-- Reviewable:end -->

otan committed:
sql: remove SURVIVE DEFAULT; rename Survive to SurvivalGoal

We still need a concept of "default" for tree.SurviveDefault so that we
do not display it syntactically, but the value will be transformed
before it is persisted.

Release note: None

ajstorm reviewed (approved):
LGTM

> Reviewed 5 of 5 files at r1, 11 of 11 files at r2.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/56837#-:-MMQMvfU-Ab_EhJG94Jj:bb6ywmx)* status: :shipit: complete! 0 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

otan commented:
thanks for the quick reviews

bors r=ajstorm
